### ISSUE TITLE
Bug in iterated algebraic extensions over finite fields

### ISSUE THREAD

**Issue by frankluebeck** [gap-system/gap#1457]
### Observed behaviour
```
f0 := GF(3);
p1 := CyclotomicPolynomial(f0, 5);
f1 := AlgebraicExtension(f0, p1);
p2 := X(f1)^5 - RootOfDefiningPolynomial(f1);
f2 := AlgebraicExtension(f1, p2);
p3 := X(f2)^5 - RootOfDefiningPolynomial(f2);
f3 := AlgebraicExtension(f2, p3);
```
crashes GAP in current master branch with the message
```
gap: cannot extend the workspace any more!
```
In GAP 4.7.6 the codes runs into a break-loop.
### Expected behaviour
The algebraic extension should be created without problems.
### Remark
The behaviour depends on the field we start with, e.g., with `f0 = GF(13)` the code works.

**Comment by hulpke** [comment]
To the best of my knowledge `AlgebraicExtension` never worked for iterated extensions. (Yes, it would be nice to have, but my existing code is too stupid.)

**Comment by frankluebeck** [comment]
@hulpke: I do not understand this comment. `AlgebraicExtension` gets a field and an irreducible polynomial over that field as argument. What could possibly go wrong with iterating this? Do you mean that `AlgebraicExtension` does not return a proper field that is valid input for iterated calls of `AlgebraicExtension`?

The reason for the bug reported above was that an intermediate field of size q <= 256, but not a GF(q), was not handled correctly by `DoImmutableMatrix` which is called within `AlgebraicExtension` (see pull request  "Fix DoImmutableMatrix for finite fields" #1481).

**Comment by fingolfin** [comment]
This works in master and even in GAP 4.10, though it works "better" in master: there, I get this:
```
gap> f3 := AlgebraicExtension(f2, p3);
<field of size 515377520732011331036461129765621272702107522001>
```
while in 4.10 I get this:
```
gap> f3 := AlgebraicExtension(f2, p3);
<algebra-with-one of dimension 5 over AsField( Field( [ a ] ), Field( [ a ] ) )>
```

So this is fixed, yay.
